Runtime/Triangulator.cs

```diff
@@ -283,49 +288,485 @@ public void Dispose()
+                }
+            }
+            public void Transform (AffineTransform2D transform, [NoAlias] NativeArray<int2> points, [NoAlias] NativeArray<int2> outPoints) {
+                if (transform != AffineTransform2D.identity) {
+                    throw new NotImplementedException();
+                }
+                outPoints.CopyFrom(points);
+            }
+            public void InverseTransform (AffineTransform2D transform, [NoAlias] NativeArray<int2> points, [NoAlias] NativeArray<int2> outPoints) {
+                if (transform != AffineTransform2D.identity) {
+                    throw new NotImplementedException();
+                }
+                outPoints.CopyFrom(points);
+            }
+
+            public int HashKey(int2 p, int2 origin, int hashSize)
+            {
+                return (int)math.floor(pseudoAngle(p.x - origin.x, p.y - origin.y) * hashSize) % hashSize;
+
+                static float pseudoAngle(float dx, float dy)
+                {
+                    var p = dx / (math.abs(dx) + math.abs(dy));
+                    return (dy > 0 ? 3 - p : 1 + p) / 4; // [0..1]
+                }
+            }
+
+            public void BoundingBox (ReadOnlySpan<int2> positions, out int2 min, out int2 max, out int2 center) {
+                min = int.MaxValue;
+                max = int.MinValue;
+                foreach (var p in positions)
+                {
+                    min = math.min(p, min);
+                    max = math.max(p, max);
+                }
+                center = (min + max) / 2;
+            }
+
+            public int2 Lerp (int2 a, int2 b, float t) {
+                return new int2(
+                    a.x + (int)math.round((b.x - a.x)*(double)t),
+                    a.y + (int)math.round((b.y - a.y)*(double)t)
+                );
+            }
+
+            double Angle(int2 a, int2 b) => math.atan2((double)Cross(a, b), (double)DotLong(a, b));
+
+            public bool SmallestInnerAngleIsBelowThreshold(int2 pA, int2 pB, int2 pC, float angleThreshold) {
+                var pAB = pB - pA;
+                var pBC = pC - pB;
+                var pCA = pA - pC;
+
+                var angles = math.float3
+                (
+                    (float)Angle(pAB, -pCA),
+                    (float)Angle(pBC, -pAB),
+                    (float)Angle(pCA, -pBC)
+                );
+
+                return math.any(math.abs(angles) < angleThreshold);
+            }
+        }
+
+        interface ICoordinateUtils<Coord, LengthSq> where Coord : unmanaged where LengthSq : unmanaged {
+            public bool IsAcuteAngle(Coord a, Coord b, Coord c);
+            public LengthSq Area2(int i, int j, int k, ReadOnlySpan<Coord> positions);
+            public Circle<Coord, LengthSq> CalculateCircumCircle(int i, int j, int k, NativeArray<Coord> positions);
+            public LengthSq CircumRadiusSq(Coord a, Coord b, Coord c);
```

Comment:
There's no need for `public` keywords in the interface body. It's simply a matter of style and can be addressed later.